### PR TITLE
Implement progress event emitting via http

### PR DESCRIPTION
**Pull request by telphan**

### CODE REVIEW

File: lib/workers/testbot.ts [hunk 1]
```diff
@@ -59,8 +60,9 @@ export class TestBot {
 	 */
 	// Firmata types devicePath as any, will do the same
 	constructor(devicePath: any, private options?: WorkerOptions) {
```
Comment (CameronDiver) [inline lib/workers/testbot.ts:62]: why is this any?

**Review by CameronDiver — APPROVED** [review]

File: lib/workers/testbot.ts (hunk 1 above)
Comment (telphan) [reply lib/workers/testbot.ts:62]: Just matching the type in firmata.

File: lib/index.ts [hunk 2]
```diff
@@ -61,15 +61,23 @@ async function setup(devicePath: string): Promise<express.Application> {
...
+			});
+
 			const timer = setInterval(() => {
 				res.write('Still Flashing');
 			}, httpServer.keepAliveTimeout);
 
 			try {
-				res.status(202);
+				worker.on('progress', progress => {
+					res.write(`progress: ${JSON.stringify(progress)}`);
```
Comment (zvin) [inline lib/index.ts]: Who will be consuming this?
Maybe
```typescript 
res.write(JSON.stringify({ progress }));
```
is better?

**Review by zvin — APPROVED** [review]

File: lib/index.ts (hunk 2 above)
Comment (telphan) [reply lib/index.ts]: The client will be consuming this, i.e. my core testing framework

**Pull request merged**